app/src/main/java/fr/neamar/kiss/CameraHandler.java

```diff
@@ -0,0 +1,105 @@
+package fr.neamar.kiss;
+
+import java.io.IOException;
+import java.util.List;
+
+import android.graphics.SurfaceTexture;
+import android.hardware.Camera;
+import android.hardware.Camera.AutoFocusCallback;
+import android.hardware.Camera.Parameters;
+import android.os.Build;
+import android.util.Log;
+import android.view.SurfaceView;
+
+public class CameraHandler {
+
+	private Camera camera = null;
+	private SurfaceTexture surfaceTexture = null;
+
+	public void openCamera() throws IOException {
+		if (camera == null) {
+			Log.d("openCamera", "Open Camera");
+			camera = Camera.open(Camera.CameraInfo.CAMERA_FACING_BACK);
```

Comment:
Method openCamera() is used only in class CameraHandler and every call is try/catched by a generic Exception (that cover also RuntimeException)